### PR TITLE
First cut at trying to unify the eclipse projects together.

### PR DESCRIPTION
**Pull request by jsuereth**
Attempt to merge "core" and "plugin" into the same project.

This just gets things working and removes some odd type duplciation for aliases.   We still aren't done cleaning, but it's at least worht a look.  cc. @benmccann

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -9,36 +9,15 @@ import com.typesafe.sbt.SbtGit._
 
 object Build extends Build {
 
-  val baseVersion = "4.0.0"
+  val baseVersion = "5.0.0"
```
Comment (benmccann) [inline project/Build.scala]: we haven't released 4.0.0 yet. just 4.0.0-RC1, so we could just call this RC2 if you wanted

Comment (jsuereth) [inline project/Build.scala]: good call

File: src/main/scala/com/typesafe/sbteclipse/core/EclipseAutoPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+import com.typesafe.sbteclipse.core.{EclipsePlugin=>EclipseCorePlugin}
+
+import scala.xml.Node
+
+/** Migrates eclipse to be an autoplugin. */
+object EclipsePlugin extends AutoPlugin {
+  override def requires = sbt.plugins.JvmPlugin
+  override def trigger = allRequirements
+  // Auto import semantics users expect today.
```
Comment (benmccann) [inline src/main/scala/com/typesafe/sbteclipse/core/EclipseAutoPlugin.scala]: I wouldn't worry about backwards compatibility too much. The only code that typically depends on this is the user's build and that's easy to change. It's when you've got another library depending on this one in between that things get stickier

**Comment by benmccann** [comment]
lgtm

File: src/main/scala/com/typesafe/sbteclipse/core/EclipseAutoPlugin.scala (hunk 2 above)
Comment (jsuereth) [inline src/main/scala/com/typesafe/sbteclipse/core/EclipseAutoPlugin.scala]: True. I've tried to preserve enough that we don't get massive breakage in existing builds, and the tests don't change.   We can be more aggressive if we want, but this should get us over the hump of a usable release.

**Comment by jsuereth** [comment]
@benmccann Woudl you want to cut an RC2 after this is merged? I can walk you through it.

**Comment by benmccann** [comment]
Sure

**Pull request merged**